### PR TITLE
added all entity definitions

### PR DESCRIPTION
**Pull request by yzummy**

### CODE REVIEW

File: server/src/app.ts [hunk 1]
```diff
@@ -63,12 +63,14 @@ if (!module.parent) {
...
 // Parse class-validator classes into JSON Schema:
+
 const metadatas = (getFromContainer(MetadataStorage) as any)
   .validationMetadatas;
 const schemas = validationMetadatasToSchemas(metadatas, {
   refPointerPrefix: '#/components/schemas/',
 });
-console.log(schemas);
+
+console.log('Here is the schema: ', schemas);
```
Comment (kevinschoonover) [inline server/src/app.ts]: Remove unnecessary console.log

File: server/src/controllers/Account.ts [hunk 2]
```diff
@@ -54,6 +51,7 @@ export class AccountController {
   @OpenAPI({
     summary: 'Update the fields of an Account associated with id',
   })
+  @ResponseSchema(Account)
```
Comment (kevinschoonover) [inline server/src/controllers/Account.ts]: For patch requests, it also takes in all of the attributes as optional parameters

File: server/src/entity/InventoryItem.ts [hunk 3]
```diff
@@ -21,34 +30,42 @@ export enum itemCategory {
...
   })
+  @IsEnum(itemCategory)
   public category: itemCategory;
 
   @Column({ length: 50, default: 'default_logo.jpg', nullable: true })
+  @IsString()
   public logoPath: string;
 
   @Column({ type: 'simple-array' })
+  @IsArray()
```
Comment (kevinschoonover) [inline server/src/entity/InventoryItem.ts]: is there any way you can specify the type of the array?

File: server/src/entity/Reservation.ts [hunk 4]
```diff
@@ -9,22 +9,27 @@ import {
...
 import { Account } from './Account';
+import { IsUUID, IsDate, IsNumber, IsArray } from 'class-validator';
 
 @Entity()
 export class Reservation extends BaseEntity {
   @PrimaryGeneratedColumn('uuid')
+  @IsUUID()
   public id: string;
 
   @CreateDateColumn()
```
Comment (kevinschoonover) [inline server/src/entity/Reservation.ts]: missing annotation

File: server/src/entity/Reservation.ts [hunk 5]
```diff
@@ -9,22 +9,27 @@ import {
...
 
   @Column({ default: 1 })
+  @IsNumber()
   public numGuests: number;
 
   @ManyToMany((type) => Account, (account) => account.reservations)
+  @IsArray()
   public guests: Account[];
 
   @ManyToOne((type) => Restaurant, (restaurant) => restaurant.reservations)
```
Comment (kevinschoonover) [inline server/src/entity/Reservation.ts]: Missing annotation

File: server/src/entity/Service.ts [hunk 6]
```diff
@@ -26,18 +27,20 @@ export enum serviceType {
...
-
   @Column({
     default: serviceType.UNKNOWN,
     enum: serviceType,
     type: 'enum',
   })
+  @IsEnum(serviceType)
   public type: serviceType;
+
+  @OneToMany((type) => Account, (account) => account.services)
```
Comment (kevinschoonover) [inline server/src/entity/Service.ts]: missing annotations

File: server/src/entity/InventoryItem.ts (hunk 3 above)
Comment (yzummy) [reply server/src/entity/InventoryItem.ts]: I tried. Cannot find it

File: server/src/controllers/Account.ts (hunk 2 above)
Comment (yzummy) [reply server/src/controllers/Account.ts]: I think it works now;.

**Pull request merged**